hydromt/io.py

```diff
@@ -615,13 +616,25 @@ def open_vector(
     gdf : geopandas.GeoDataFrame
         Parsed geometry file
     """
-    filtered = False
+
+    def _read(f: pyio.IOBase) -> gpd.GeoDataFrame:
+        bbox_reader = gis_utils.prepare_pyogrio_reader_filters(f, bbox, geom, crs)
+        f.seek(0)
```

Comment:
Why the seek?

Reply:
This function partially exhausts the `f`, as it reads metadata using `pyogrio.read_info`.